### DISCUSSION TITLE
Live Transcription Ready State Connection Latency

### DISCUSSION THREAD

**Discussion by ReynoldHoule** [deepgram/community#880] [General help] [answered]
I've noticed intermittent latency and connectivity issues when opening a live transcription socket, starting from July 25, 2024. We've observed several instances where the connection remains stuck in the "connecting" status. This issue seems to align with the incident reported on July 30th: [Deepgram Status Incident](https://status.deepgram.com/incidents/lq6bmxr5097r).

Although the incident report indicates that the issue has been resolved, we continue to experience these intermittent problems. Are there any ongoing issues, or are others still encountering similar challenges? Despite these occurrences, we have had multiple successful live transcription runs as well.

`  Using Deepgram SDK

    const deepgram = new Deepgram(deepgramApiKey);

    const deepgramOptions: LiveTranscriptionOptions = {
      smart_format: true,
      language: 'en-US',
      model: 'nova-2-ea',
      profanity_filter: true,
      interim_results: true,
      punctuate: true,
      paragraphs: true,
    };

    // this initializes an open socket to deepgram
    // it does take a time to connect
    // we can check if the connection is open by deepgramLive.getReadyState() == 1
    const deepgramLive = deepgram.transcription.live(deepgramOptions);
`

`Sending Data to Deepgram, data comes from live stream - we are seeing the state as "CONNECTING" here and also when trying to keep the socket alive

      // Process audio data here if deepgram connection is established
      if (deepgramLive.getReadyState() === 1) {
        // send audio data to deepgram
        deepgramLive.send(data);
      }
        
`

`Keep Alive Check, ran on intervals until we have a live feed coming in to keep the connection open.
        
      // Check if the WebSocket is open before sending keep-alive messages
      if (deepgramLive.getReadyState() === 1) {
        deepgramLive.keepAlive();
      }`

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @ReynoldHoule, we chatted via email, but to summarize our discussion for the benefit of the community here on GitHub:
 
Your description does sound like it could be related to our published incident. We have continued to manage high levels of streaming traffic in recent days. 
 
To better manage your connections, I recommend waiting up to 5 seconds to establish a connection, and waiting 1-2 seconds before retrying. Our DNS records TTL is 10s, and typically we see one datacenter more affected than others during periods of streaming latency, so chances are you could easily hit a less-overloaded datacenter after 3-4 retries.
